### PR TITLE
Socket address in protobuf

### PR DESCRIPTION
**Pull request by Keksoj**
two types are used:

- `std::net::SocketAddr` for the whole proxying logic, storing the state, listening on TCP sockets, etc
- `sozu_command_lib::proto::command::SocketAddress` (note the full-word ending here) for *transmitting* socket addresses between clients and main process, between main process and workers

### CODE REVIEW

File: command/state.json
Comment (FlorentinDUBOIS) [inline command/state.json:1]: Should be added to the .gitignore :)

Comment (Keksoj) [reply command/state.json:1]: oh shoot! I'm removing this.

File: lib/src/https.rs [hunk 1]
```diff
@@ -861,13 +854,10 @@ impl HttpsProxy {
         let len = self.listeners.len();
 
         self.listeners
-            .retain(|_, listener| listener.borrow().address.to_string() != remove.address);
+            .retain(|_, listener| listener.borrow().address != remove.address.clone().into());
```
Comment (FlorentinDUBOIS) [inline lib/src/https.rs:857]: Is this clone necessary?

**Review by FlorentinDUBOIS — CHANGES_REQUESTED** [review]
Seems ok?

File: lib/src/https.rs (hunk 1 above)
Comment (Keksoj) [reply lib/src/https.rs:854]: TL;DR: yes

We looked into Into/From, they takes ownership no matter what. Implementing them on a reference proves to be tricky.

We think the compiler is probably smart enough to spare the stack's space when "copying" while clonig and reusing memory.

We could implement our own version of From that would not take ownership, but it's 32 bytes of data we're talking about here.
Although a conversion from SocketAddr to the protobuf format will needed at each access log emission, converting to this binary format will still be way more efficient than performing a `to_string()`.
The most crucial performance bottlenecks lie elsewhere, in TLS traffic for instance.

Comment (FlorentinDUBOIS) [reply lib/src/https.rs:854]: I wonder about something like this is feasable?
```suggestion
let remove_address: SocketAddress = remove.address.clone().into();
...
            .retain(|_, listener| listener.borrow().address != remove_address);
```

Comment (Keksoj) [reply lib/src/https.rs:854]: sure can do

Comment (Keksoj) [reply lib/src/https.rs]: done!

**Review by FlorentinDUBOIS — APPROVED** [review]

**Pull request merged**